module/zfs/arc.c

```diff
@@ -5364,7 +5370,10 @@ arc_tuning_update(void)
 		arc_c_max = zfs_arc_max;
 		arc_c = arc_c_max;
 		arc_p = (arc_c >> 1);
-		arc_meta_limit = (3 * arc_c_max) / 4;
+		/* Valid range of arc_meta_limit: arc_meta_min - arc_c_max */
+		arc_meta_limit = (arc_c_max * zfs_arc_meta_limit_percent) / 100;
+		arc_meta_limit = MIN(arc_meta_limit, arc_c_max);
+		arc_meta_limit = MAX(arc_meta_min, arc_meta_limit);
```

Comment:
Another way to structure this which I think is a little more readable is to use a local variable to store the percentage after performing the bounds check.  For example, something like this which includes @dweeezil's proposed `zfs_arc_dnode_limit_percent` which is a good idea.

```diff
diff --git a/module/zfs/arc.c b/module/zfs/arc.c
index cc26b2e..d3e59dd 100644
--- a/module/zfs/arc.c
+++ b/module/zfs/arc.c
@@ -5357,6 +5357,9 @@ arc_state_multilist_index_func(multilist_t *ml, void *obj)
 static void
 arc_tuning_update(void)
 {
+       int percent;
+
        /* Valid range: 64M - <all physical memory> */
        if ((zfs_arc_max) && (zfs_arc_max != arc_c_max) &&
            (zfs_arc_max > 64 << 20) && (zfs_arc_max < ptob(physmem)) &&
@@ -5364,8 +5367,10 @@ arc_tuning_update(void)
                arc_c_max = zfs_arc_max;
                arc_c = arc_c_max;
                arc_p = (arc_c >> 1);
-               arc_meta_limit = (3 * arc_c_max) / 4;
-               arc_dnode_limit = arc_meta_limit / 10;
+               percent = MAX(MIN(zfs_arc_meta_limit_percent, 100), 0);
+               arc_meta_limit = (percent * arc_c_max) / 100;
+               percent = MAX(MIN(zfs_arc_dnode_limit_percent, 100), 0);
+               arc_dnode_limit = (percent * arc_meta_limit) / 100;
        }
 
        /* Valid range: 32M - <arc_c_max> */
```